pkg/clusterapi/aws/actuator_test.go

```diff
@@ -0,0 +1,338 @@
+		Filters: []*ec2.Filter{{Name: aws.String("tag:Name"), Values: []*string{&vpcName}}},
+	}
+	describeVpcsOutput := ec2.DescribeVpcsOutput{
+		Vpcs: []*ec2.Vpc{
+			{
+				VpcId: aws.String(vpcID),
+			},
+		},
+	}
+	mockAWSClient.EXPECT().DescribeVpcs(&describeVpcsInput).Return(&describeVpcsOutput, nil)
+}
+
+func addDescribeSubnetsMock(mockAWSClient *MockAWSClient, az, vpcID, subnetID string) {
+	input := ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			{Name: aws.String("vpc-id"), Values: []*string{aws.String(vpcID)}},
+			{Name: aws.String("availability-zone"), Values: []*string{aws.String(az)}},
+		},
+	}
+	output := ec2.DescribeSubnetsOutput{
+		Subnets: []*ec2.Subnet{
+			{
+				SubnetId: aws.String(subnetID),
+			},
+		},
+	}
+	mockAWSClient.EXPECT().DescribeSubnets(&input).Return(&output, nil)
+}
+
+func addDescribeSecurityGroupsMock(mockAWSClient *MockAWSClient, vpcID, vpcName string, isMaster, isInfra bool) {
+	// Using the input builder from the production code. Behavior of this function is tested separately.
+	input := buildDescribeSecurityGroupsInput(vpcID, vpcName, isMaster, isInfra)
+	output := ec2.DescribeSecurityGroupsOutput{
+		SecurityGroups: []*ec2.SecurityGroup{},
+	}
+	// Using the explicit index where we store our name filters, may want to search for it if this
+	// proves too frail.
+	// Add a fake ID for each security group we're searching for.
+	for _, groupName := range input.Filters[1].Values {
+		output.SecurityGroups = append(output.SecurityGroups, &ec2.SecurityGroup{GroupId: aws.String(fmt.Sprintf("%s-ID", *groupName))})
+	}
+	mockAWSClient.EXPECT().DescribeSecurityGroups(input).Return(&output, nil)
+}
+
+// testClusterDeployment creates a new test ClusterDeployment
+func testClusterDeployment() *clustopv1.ClusterDeployment {
+	clusterDeployment := &clustopv1.ClusterDeployment{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:       testClusterDeploymentUUID,
+			Name:      testClusterDeploymentName,
+			Namespace: testNamespace,
+		},
+		Spec: clustopv1.ClusterDeploymentSpec{
+			MachineSets: []clustopv1.ClusterMachineSet{
+				{
+					ShortName: "",
+					MachineSetConfig: clustopv1.MachineSetConfig{
+						Infra:    true,
+						Size:     1,
+						NodeType: clustopv1.NodeTypeMaster,
+					},
+				},
+				{
+					ShortName: "compute",
+					MachineSetConfig: clustopv1.MachineSetConfig{
+						Infra:    false,
+						Size:     1,
+						NodeType: clustopv1.NodeTypeCompute,
+					},
+				},
+			},
+			Hardware: clustopv1.ClusterHardwareSpec{
+				AWS: &clustopv1.AWSClusterSpec{
+					SSHUser:     "clusteroperator",
+					Region:      testRegion,
+					KeyPairName: "libra",
+				},
+			},
+			ClusterVersionRef: clustopv1.ClusterVersionReference{
+				Name:      testClusterVerName,
+				Namespace: testClusterVerNS,
+			},
+		},
+	}
+	return clusterDeployment
+}
+
+func testCluster() *capiv1.Cluster {
+	clusterDeployment := testClusterDeployment()
+	cluster, _ := clusterdeployment.BuildCluster(clusterDeployment)
```

Comment:
Don't swallow this error. Pass in the `*testing.T` and fail the test if the cluster cannot be built.